src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -933,9 +1673,594 @@ bool W3DDisplay::Is_LetterBox_Fading()
+        m_displayStrings[4]->Set_Text(str1);
+
+        lastUpdateTime64 = PerformanceCounter.QuadPart;
+        frameTime = 0.0;
+        frameCount = 0;
+        drawCalls = 0;
+        sortingPolygons = 0;
+
+        str1.Format(L"3-Way Blends: %d/%d, Shoreline Blends: %d/%d",
+            g_theTerrainRenderObject->Get_Num_Extra_Blend_Tiles(true),
+            g_theTerrainRenderObject->Get_Num_Extra_Blend_Tiles(false),
+            g_theTerrainRenderObject->Get_Num_Water_Blend_Tiles(true),
+            g_theTerrainRenderObject->Get_Num_Water_Blend_Tiles(false));
+        m_displayStrings[15]->Set_Text(str1);
+
+        Coord3D pos;
+        g_theTacticalView->Get_Position(&pos);
+        str1.Format(L"Camera zoom: %g, pitch: %g/%g, yaw: %g, pos: %g, %g, %g, FOV: %g\n"
+                    L"       Height above ground: %g Terrain height: %g",
+            g_theTacticalView->Get_Zoom(),
+            g_theTacticalView->Get_Pitch(),
+            g_theTacticalView->Get_FX_Pitch(),
+            g_theTacticalView->Get_Angle(),
+            pos.x,
+            pos.y,
+            pos.z,
+            g_theTacticalView->Get_Field_Of_View(),
+            g_theTacticalView->Get_Current_Height_Above_Ground(),
+            g_theTacticalView->Get_Terrain_Height_Under_Camera());
+        m_displayStrings[5]->Set_Text(str1);
+
+        str1.Format(L"States: ");
+
+        if (g_theKeyboard->Is_Shift()) {
+            str1.Concat(L"Shift(");
+
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_LSHIFT) != 0) {
+                str1.Concat(L"L");
+            }
+
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_RSHIFT) != 0) {
+                str1.Concat(L"R");
+            }
+
+            str1.Concat(L") ");
+        }
+
+        if (g_theKeyboard->Is_Ctrl()) {
+            str1.Concat(L"Ctrl(");
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_LCTRL) != 0) {
+                str1.Concat(L"L");
+            }
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_RCTRL) != 0) {
+                str1.Concat(L"R");
+            }
+            str1.Concat(L") ");
+        }
+
+        if (g_theKeyboard->Is_Alt()) {
+            str1.Concat(L"Alt(");
+
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_LALT) != 0) {
+                str1.Concat(L"L");
+            }
+
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_RALT) != 0) {
+                str1.Concat(L"R");
+            }
+            str1.Concat(L") ");
+        }
+
+        MouseIO *status = g_theMouse->Get_Mouse_Status();
+
+        if (status->left_state) {
+            str1.Concat(L"LMB ");
+        }
+
+        if (status->middle_state) {
+            str1.Concat(L"MMB ");
+        }
+
+        if (status->right_state) {
+            str1.Concat(L"RMB ");
+        }
+
+        const Object *object = nullptr;
+        Drawable *d;
```

Comment:
Move this into each #ifdef body please. `Drawable *drawable = `

Reply:
fixed.